docs/releases/3.1.txt

```diff
@@ -452,6 +452,50 @@ Miscellaneous
 Backwards incompatible changes in 3.1
 =====================================
 
+Model field inheritance
+-----------------------
+
+Model field descriptors will now always be set on the child class when
+overriding an attribute on a parent class. Previously, a truthy attribute on
+the parent class would inhibit the descriptor from being set.
+
+When overriding fields from an abstract parent class, this change should be
+transparent, except in circumstances of diamond-shaped multiple inheritance.
+The inherited field resolution in such cases is strictly depth-first, rather
+than Python MRO depth-first-with-pull-up for diamond-shaped inheritance. This
+difference is in-line with the documented :ref:`limitations on model field
+inheritance <field-name-hiding>`.
+
+To account for this change, you may need to push-down model field declarations
+to child classes if your models have complex diamond-shaped inheritance
+structures. For example, if you have the following kind of structure in your
+models::
+
+    class Base(models.Model):
+        name = models.CharField(max_length=30)
+
+        class Meta:
+            abstract = True
+
+    class ParentOne(Base):
+        class Meta:
+            abstract = True
+
+    class ParentTwo(Base):
+        name = models.CharField(max_length=50)
+
+        class Meta:
+            abstract = True
+
+    class Derived(ParentOne, ParentTwo):
+        pass
```

Comment:
This example raises:
```
test_one.Derived.name: (models.E006) The field 'name' clashes with the field 'name' from model 'test_one.derived'.
```

Reply:
Good catch! 

I ran this exact test **but** defining the model classes dynamically in the test case. I want to check adding similar to a `models.py` to make sure/see what happens when the checks are run. 

 I need to think about this one a bit. Ref's https://github.com/django/django/pull/11337#discussion_r391534297